File: src/main/java/pl/mroz/buddiesapi/infrastructure/database/rental/RentalRepositoryConfig.java
package pl.mroz.buddiesapi.infrastructure.database.rental;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import pl.mroz.buddiesapi.domain.rental.RentalRepository;

@Configuration
public class RentalRepositoryConfig {

    @Bean
    RentalRepository rentalRepository(RentalRepositoryImpl.RentalJpaRepository jpaRepository) {
        return new RentalRepositoryImpl(jpaRepository);
    }
}

File: src/main/kotlin/buddiesapi/infrastructure/database/rental/RentalRepositoryConfig.kt
package buddiesapi.infrastructure.database.rental

import buddiesapi.domain.rental.RentalRepository
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class RentalRepositoryConfig {
    @Bean
    fun rentalRepository(jpaRepository: RentalRepositoryImpl.RentalJpaRepository): RentalRepository {
        return RentalRepositoryImpl(jpaRepository)
    }
}
